### PR TITLE
(fix) Isomorphic Git::Lightning FS: Update to Latest API Design

### PR DESCRIPTION
**Pull request by 0xTheProDev**
Discussion Thread: https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/58917

Signed-off-by: Progyan Bhattacharya <progyan@interviewkickstart.com>

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by Progyan1997** [comment]
@[jesseditson](https://github.com/jesseditson) Can you take a quick look here?

**Review by jesseditson — APPROVED** [review]
Nice! I haven't manually tested (I likely won't have time until at least next week), but looking at the code it looks good to me!

File: types/isomorphic-git__lightning-fs/isomorphic-git__lightning-fs-tests.ts [hunk 1]
```diff
@@ -4,8 +4,8 @@ const FS = new LightningFS('FS');
 
 const originalData = new Uint8Array([138, 132]);
 
-FS.writeFile('/sample.txt', originalData).then(() => {
-  FS.readFile('./sample.txt').then((data: Uint8Array) => {
+FS.writeFile('/sample.txt', originalData, undefined, () => {
+  FS.readFile('./sample.txt', undefined, (_err: Error, data: string | Uint8Array) => {
```
Comment (jesseditson) [inline types/isomorphic-git__lightning-fs/isomorphic-git__lightning-fs-tests.ts:8]: Seems weird that this is different than the [node.js type](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/d7d9c1d76dead09380e702d858b40432a29d8a48/types/node/fs.d.ts#L2424) (the second arg should be optional), but the upstream types also are defined this way: https://github.com/isomorphic-git/lightning-fs/blob/6f3ae38eefdd2e2b0135e90b57e44bf29ecdb5bb/index.d.ts#L50 - so this is probably not worth over-defining on this side.

**Review by jesseditson — APPROVED** [review]
LGTM!

**Comment by Progyan1997** [comment]
Ready to merge

**Pull request merged**